core/types/transaction_signing.go

```diff
@@ -40,7 +40,7 @@ type sigCache struct {
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int, blockTime uint64) Signer {
 	var signer Signer
 	switch {
-	case config.IsCancun(blockNumber, blockTime):
+	case config.IsCancun(blockNumber, blockTime) && !config.IsOptimism():
```

Comment:
This means that you cannot use `op-geth` to build txs to send to L1 if they are blob txs unless you very explicitly use `NewCancunSigner`

Reply:
It doesn't, you just can't use op-geth with an Optimism chain config for L1 txs. If you use a regular L1 chain config without optimism active, then you can create L1 txs just fine.